Content.Shared/RCD/RCDPrototype.cs

```diff
@@ -0,0 +1,120 @@
+/// </summary>
+[Prototype("rcd")]
+public sealed class RCDPrototype : IPrototype
+{
+    [IdDataField]
+    public string ID { get; private set; } = default!;
+
+    /// <summary>
+    /// The RCD mode associated with the operation
+    /// </summary>
+    [DataField(required: true), ViewVariables(VVAccess.ReadOnly)]
+    public RcdMode Mode { get; private set; } = RcdMode.Invalid;
+
+    /// <summary>
+    /// The name associated with the prototype
+    /// </summary>
+    [DataField("name"), ViewVariables(VVAccess.ReadOnly)]
+    public string SetName { get; private set; } = "Unknown";
+
+    /// <summary>
+    /// The name of the radial container that this prototype will be listed under on the RCD menu
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadOnly)]
+    public string Category { get; private set; } = "Undefined";
+
+    /// <summary>
+    /// Texture path for this prototypes menu icon
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadOnly)]
+    public SpriteSpecifier? Sprite { get; private set; } = null;
+
+    /// <summary>
+    /// The entity prototype that will be constructed (mode dependent)
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadOnly)]
+    public string? Prototype { get; private set; } = string.Empty;
+
+    /// <summary>
+    /// Number of charges consumed when the operation is completed
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadOnly)]
+    public int Cost { get; private set; } = 1;
+
+    /// <summary>
+    /// The length of the operation 
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadOnly)]
+    public float Delay { get; private set; } = 1f;
+
+    /// <summary>
+    /// The visual effect that plays during this operation
+    /// </summary>
+    [DataField("fx"), ViewVariables(VVAccess.ReadOnly)]
+    public EntProtoId? Effect { get; private set; } = null;
+
+    /// <summary>
+    /// A list of rules that govern where the entity prototype can be contructed
+    /// </summary>
+    [DataField("rules"), ViewVariables(VVAccess.ReadOnly)]
+    public HashSet<RcdConstructionRule> ConstructionRules { get; private set; } = new();
+
+    /// <summary>
+    /// The collision mask used for determining whether the entity prototype will fit into a target tile
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadOnly)]
+    public CollisionGroup CollisionMask { get; private set; } = CollisionGroup.None;
+
+    /// <summary>
+    /// Specifies a set of custom collision bounds for determining whether the entity prototype will fit into a target tile 
+    /// </summary>
+    /// <remarks>
+    /// Should be set assuming that the entity faces south.
+    /// Make sure that Rotation is set to RcdRotation.User if the entity is to be rotated by the user
+    /// </remarks>
+    [DataField, ViewVariables(VVAccess.ReadOnly)]
+    public Box2? CollisionBounds { get; private set; } = null;
+
```

Comment:
Should this be IPhysShape?

Reply:
I don't need to actually simulate any physics, I just need to check if two entities overlap